http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/KubernetesServiceDbQueries.scala

```diff
@@ -43,17 +44,16 @@ object KubernetesServiceDbQueries {
     for {
       clusterOpt <- kubernetesClusterQuery.getMinimalActiveClusterByName(saveKubernetesCluster.googleProject)
 
-      failure = DBIO.failed(
-        KubernetesAppCreationException(
-          s"You cannot create an app while a cluster is in ${KubernetesClusterStatus.Precreating} or ${KubernetesClusterStatus.Provisioning}"
-        )
-      )
       eitherClusterOrError <- clusterOpt match {
         case Some(cluster) =>
           cluster.status match {
-            case KubernetesClusterStatus.Precreating  => failure
-            case KubernetesClusterStatus.Provisioning => failure
-            case _                                    => DBIO.successful(SaveClusterResult(cluster, true))
+            case s if KubernetesClusterStatus.creatingStatuses contains s =>
+              DBIO.failed(
```

Comment:
Minor, but I wonder if this should be handled at the service layer (e.g. `LeoKubernetesServiceInterp`) instead of the DB layer since it feels more like business logic instead of something that should be enforced in the DB.

Reply:
Yeah, I can see the argument for it. I view these `ServiceDbQueries` as database business logic for the service, so I like having the low-level logic here to keep the flow in the ServiceInterp simplified